library/asb_set_state.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/python
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.0',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: asb_set_state
+short_description: Sets state in a ConfigMap
+description:
+     - Provides a secure method for allowing Service Bundle developers
+       to set state during a Service Bundle action.
+       Provided key, value pair is stored in a ConfigMap for later retrieval.
+author:
+    - "Red Hat, Inc."
+options:
+    key:
+        description:
+            - Unique key
+        required: true
+    value:
+        description:
+            - Literal value
+        required: true
+ env:
+        # Set via the downward API on the APB Pod
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+'''
+
+EXAMPLES = '''
+- name: Set some example state
+  asb_set_state:
+    key: "example-key"
+    value: "some-value"
+'''
+
+import os
+
+from ansible.module_utils.basic import AnsibleModule
+try:
+    from kubernetes import client, config
+    from kubernetes.client.rest import ApiException
+    config.load_kube_config()
```

Comment:
Not necessarily a blocker, but this will only work if the user has a valid kubeconfig, and won't allow other supported authentication types such as incluster/token/user+pass. Here's how the 2.6 k8s modules handle the various authentication types: https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/k8s/common.py#L124

Reply:
Sure yeah. I think in this PR I'll just keep to the format of the existing modules and we can refactor them all to support this later if needed. Thanks